fix typo and common mistakes (#18998)
matplotlib/matplotlib: doc/faq/howto_faq.rst

as `datetime.datetime` objects.

If you prefer the pandas converters and locators, you can register them.  This

> is done automatically when calling a pandas plot function and maybe

Reviewer: ```suggestion
is done automatically when calling a pandas plot function and may be
```

https://www.google.ca/amp/s/dictionary.cambridge.org/amp/british-grammar/maybe-or-may-be

recursively search the artist for any artists it may contain that meet
some criteria (e.g., match all :class:`~matplotlib.lines.Line2D`
instances or match some arbitrary filter function).  For example, the

> the following snippet finds every object in the figure which has a

Reviewer: ```suggestion
following snippet finds every object in the figure which has a
```
Repeated the

The :meth:`~matplotlib.pyplot.savefig` command has a keyword argument
*transparent* which, if 'True', will make the figure and axes
backgrounds transparent when saving, but will not affect the displayed

> the image on the screen.

Reviewer: ```suggestion 
image on the screen.
```

Author: I think these errors are due to various English standards.

Reviewer: I don't think there is any version of English where inserting an extra determinant between an adjective and its noun is considered correct.

Reviewer 2: "... will not affect the displayed image on the screen."
versus
"... will not affect the displayed the image on the screen."

The former is correct. Perhaps the split line is causing confusion?

If you need finer-grained control, e.g., you do not want full transparency
or you want to affect the screen displayed version as well, you can set

> the alpha properties directly.  The figure has a:class:`~matplotlib.patches.Rectangle` instance called *patch*

Reviewer 2: Missing space after `a`. And that part after `a` should probably go back on its own line.

> and the axes have a Rectangle instance called *patch*.  You can set

Reviewer 2: This might be a result of confusion around MPL's use of the word "axes". In this case, it is ambiguous whether we mean it in the singular sense (as in, axes instance), or the plural sense (multiple axes objects).

.. _howto-subplots-adjust:

> Move the edge of axes to make room for tick labels

Reviewer 2: Again, possible ambiguity around singular or plural axes.

In most use cases, it is enough to simply change the subplots adjust
parameters as described in :ref:`howto-subplots-adjust`.  But in some
cases, you don't know ahead of time what your tick labels will be, or
how large they will be (data and labels outside your control maybe

> being fed into your graphing application), and you may need to

Reviewer 2: In either case, the wording here is awkward. "may be being" and "maybe being" are both poor wording, so this should just get reworded entirely, I think.

----------------------------------

> Within axes, the order that the various lines, markers, text,

Reviewer 2: I like the original phrasing better here, as it points towards the singular form, but perhaps "an axes instance" might be better?

-----------------------------------

When you want to view your plots on your display,

> the user interface backend will need to start the GUI main loop.

Reviewer 2: Will someone who is a GUI expert weigh in on the correct nomenclature?

Reviewer 3: I think splitting is correct (https://en.wikipedia.org/wiki/Event_loop).

.. note::
   Because closing a figure window unregisters it from pyplot, you must call

>    `~matplotlib.pyplot.savefig` *before* calling ``show`` if you wish to save the figure as well as view it.

Reviewer 2: Do we have line-length limits on .rst files?

Reviewer 3: It's not strictly enforced because you sometimes need longer lines, e.g. for links or larger tables. However, you should wrap at 79 chars if possible.

However, it is *possible* to force Matplotlib to draw after every command,
which might be what you want when working interactively at the
python console (see :ref:`mpl-shell`), but in a script, you want to
defer all drawing until the call to ``show``.  This is especially
important for complex figures that take some time to draw.
:func:`~matplotlib.pyplot.show` is designed to tell Matplotlib that
you're all done issuing commands and you want to draw the figure now.

.. note::

    :func:`~matplotlib.pyplot.show` should typically only be called at most once per script and it should be the last line of your script.  At that point, the GUI takes control of the interpreter.

>     If you want to force a figure drawing, use

Reviewer 2: Similar question with line-length limits in .rst files.
